featurebyte/query_graph/node/base.py

```diff
@@ -825,6 +942,33 @@ def _derive_sdk_code(
         left_operand, right_operand = self._reorder_operands(left_operand, right_operand)
         return statements, ExpressionStr(self.generate_expression(left_operand, right_operand))
 
+    def _derive_on_demand_view_code(
```

Comment:
This function seems almost identical to `_derive_sdk_code`, wondering if this is common and possible to refactor and reuse more code? Something to be considered later.

Reply:
Good point, should refactor here.